include/TPP/IR/StructuredOpMatcher.h

```diff
@@ -95,11 +95,31 @@ struct HasMap {
 };
 
 // Callble object to verify if `map` is a projected permutation map.
+// We require the dimensions to be in sorted order this avoid filtering
```

Comment:
`isProjectedPermutation` is not enough for our use case as we need to make sure the dimensions are broadcastable. For example, we should disallow a transpose add operation with map:
affine_map<(d0, d1) -> (d1, d0)> on input and affine_map<(d0, d1) -> (d0, d1)> on output.